allofplos/corpus_class.py

```diff
@@ -0,0 +1,267 @@
+from collections import OrderedDict
+import datetime
+import json
+from pathlib import Path
+import os
+import random
+
+from tqdm import tqdm
+
+from . import corpusdir
+
+from .article_class import Article
+from .corpus_helpers import hash_file
+from .gdrive import (get_zip_metadata, download_file_from_google_drive,
+                     unzip_articles, zip_id, local_zip)
+from .transformations import filename_to_doi, doi_to_path
+
+hash_json = 'corpus_hash.json'
+
+
+class Corpus():
+    """A collection of PLOS articles."""
+
+    def __init__(self, directory=corpusdir, plos_network=False, extension='.xml'):
+        """Creation of an article corpus class."""
+        self.directory = directory
+        self.plos_network = plos_network
+        self.extension = extension
+
+    @property
+    def directory(self):
+        """
+        :returns: directory of corpus
+        :rtype: {str}
+        """
+        return self._directory
+
+    @directory.setter
+    def directory(self, d):
+        """
+        Reset memoized info when changing the path to the corpus.
+        """
+        self._directory = d
```

Comment:
This should check that it actually is a directory first. 

If at any point you specify a spec for these corpus directories, it should also check to make sure that spec is followed.